spec/consensus/gossip/files.md

```diff
@@ -0,0 +1,125 @@
+# Files and Types
+
+Files and types related to the State/Gossip interaction, excluding testing related.
+
+## reactor.go
+- Entry point for the consensus reactor (State+Gossip)
+    - `NewReactor`
+    - `OnStart`
+    - `OnStop`
+    - `SwitchToConsensus`
+
+- Implements the Gossip Layer
+    - Peer management
+        - `InitPeer`
+        - `AddPeer`
+        - `RemovePeer`
+
+    - Message delivery
+        - `Receive`
+            - Processes messages delivered by the Gossip layer and updates consensus (`State`) and peer information (`PeerState`)
+            - Messages are received through multiple channels
+                - `StateChannel`: `NewRoundStepMessage`, `NewValidBlockMessage`, `HasVoteMessage`, `VoteSetMaj23Message`
+                - `DataChannel`: `ProposalMessage`, `ProposalPOLMessage`, `BlockPartMessage`
+                - `VoteChannel`: `VoteMessage`
+                - `VoteBitsChannel`: `VoteSetBitsMessage`
+
+        - `PeerState` update
+            - `ApplyNewRoundStepMessage`
+            - `ApplyNewValidBlockMessage`
+            - `ApplyHasVoteMessage`
+            - `SetHasProposal`
+            - `ApplyProposalPOLMessage`
+            - `SetHasProposalBlockParts`
+            - `SetHasVote`
+            - `ApplyVoteSetBitsMessage`
+
+        - Peer information requests
+            - `VoteSetMaj23Message` (CODE TODO: Move handling from receive to helper function.)
+
+    - [Consensus state](#state.go) update
```

Comment:
More specifically, messages are delivered to the consensus implementation, via the shared `peerMsgQueue`.